cpp/uuid_module/uuid_module.cpp

```diff
@@ -11,14 +11,13 @@ constexpr char const *kFieldUuid = "uuid";
 
 void Generate(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
   try {
-    uuid_t id;
-    uuid_generate(id);
-    char *string = new char[100];
-    uuid_unparse(id, string);
+    std::random_device dev;
+    std::mt19937 rng(dev());
+    auto id = uuids::uuid_random_generator{rng}();
+    auto id_str = uuids::to_string(id);
 
     auto *record = mgp::result_new_record(result);
-
-    mg_utility::InsertStringValueResult(record, kFieldUuid, string, memory);
+    mg_utility::InsertStringValueResult(record, kFieldUuid, id_str.c_str(), memory);
```

Comment:
I think this is a bug here. 
auto id_str will be a local string variable. 
And 
`mg_utility::InsertStringValueResult(record, kFieldUuid, id_str.c_str(), memory);`
will return a pointer to the local variable which will yield undefined behavior, right? 

@BorisTasevski @Josipmrden

Reply:
Actually no error problem here, since underneath we use mgp::value_make_string function which creates new string from this object which is alive still, so all good.